src/ILCompiler.Compiler/src/Compiler/MetadataGeneration.cs

```diff
@@ -48,6 +48,10 @@ public class MetadataGeneration
 
         internal NativeLayoutInfoNode NativeLayoutInfo { get; private set; }
         internal GenericsHashtableNode GenericsHashtable { get; private set; }
+        internal ExactMethodInstantiationsNode ExactMethodInstantiations { get; private set; }
```

Comment:
Please don't make ExactMethodInstantiations and CommonFixupsTable properties. These are mutable nodes and we should be extremely careful about handing out references to things that are mutable. Having mutable state exposed to all of the compiler makes this hard to reason about.

The CommonFixupsTable is being explicitly handed out to the nodes that MetadataGeneration knows will need and use responsibly. We should also find a way to make NativeReferencesTable private.